src/main/java/uk/gov/companieshouse/api/accounts/controller/PreviousPeriodController.java

```diff
@@ -112,6 +114,48 @@ public ResponseEntity get(@PathVariable("companyAccountId") String companyAccoun
         return apiResponseMapper.mapGetResponse(responseObject.getData(), request);
     }
 
+    @PutMapping
+    public ResponseEntity update(@RequestBody @Valid PreviousPeriod previousPeriod,
+            BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+            HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.PREVIOUS_PERIOD.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Errors errors = previousPeriodValidator.validatePreviousPeriod(previousPeriod);
+        if (errors.hasErrors()) {
+            LOGGER.error( "Previous period validation failure");
+            logValidationFailureError(getRequestId(request), errors);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
```

Comment:
Shall we be checking if the transaction is null?  same as you did with the `Smallfull`?
```
 SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
        if (smallFull.getLinks().get(SmallFullLinkType.PREVIOUS_PERIOD.getLink()) == null) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
```

Reply:
Sorry, I just seen this after I merged, but I believe the transaction will never be null as the interceptor populates the request with a transaction and if it doesn't it returns an error at the interceptor.